nipype/pipeline/engine/nodes.py

```diff
@@ -277,23 +274,29 @@ def hash_exists(self, updatehash=False):
         # of the dictionary itself.
         hashed_inputs, hashvalue = self._get_hashval()
         outdir = self.output_dir()
-        if op.exists(outdir):
-            logger.debug('Output dir: %s', to_str(os.listdir(outdir)))
-        hashfiles = glob(op.join(outdir, '_0x*.json'))
-        logger.debug('Found hashfiles: %s', to_str(hashfiles))
-        if len(hashfiles) > 1:
-            logger.info(hashfiles)
-            logger.info('Removing multiple hashfiles and forcing node to rerun')
-            for hashfile in hashfiles:
-                os.unlink(hashfile)
         hashfile = op.join(outdir, '_0x%s.json' % hashvalue)
-        logger.debug('Final hashfile: %s', hashfile)
-        if updatehash and op.exists(outdir):
-            logger.debug("Updating hash: %s", hashvalue)
-            for file in glob(op.join(outdir, '_0x*.json')):
-                os.remove(file)
-            self._save_hashfile(hashfile, hashed_inputs)
-        return op.exists(hashfile), hashvalue, hashfile, hashed_inputs
+        logger.debug('Node hash value: %s', hashvalue)
+
+        if op.exists(outdir):
+            # Find previous hashfiles
+            hashfiles = glob(op.join(outdir, '_0x*.json'))
+            if len(hashfiles) > 1:  # Remove hashfiles if more than one found
+                logger.info('Removing hashfiles (%s) and forcing node to rerun',
+                            ', '.join(['"%s"' % op.basename(h) for h in hashfiles]))
+                for hf in hashfiles:
+                    os.remove(hf)
+
+            if updatehash and len(hashfiles) == 1:
```

Comment:
i don't think the second clause is necessary or the os.remove statement.

the intent of `updatehash==True` was to always update the hash and return true. not just when there was only one hashfile. it was intended to be used cautiously.